integrations/pagerduty/main.py

```diff
@@ -28,17 +31,23 @@ def initialize_client() -> PagerDutyClient:
 async def enrich_service_with_analytics_data(
     client: PagerDutyClient, services: list[dict[str, Any]], months_period: int
 ) -> list[dict[str, Any]]:
-    analytics_data = await asyncio.gather(
-        *[
-            client.get_service_analytics(service["id"], months_period)
-            for service in services
-        ]
-    )
+    enriched_services = []
+
+    for i in range(0, len(services), ANALYTICS_BATCH_SIZE):
+        batch = services[i : i + ANALYTICS_BATCH_SIZE]
+        analytics_data = await asyncio.gather(
+            *[
+                client.get_service_analytics(service["id"], months_period)
+                for service in batch
+            ]
+        )
```

Comment:
do we really need this if we have semaphore and batch of services?

Reply:
Not. It is redundant with the semaphore and rate limiting being handled by the client.